lib/sync_checker/checks/topics_check.rb

```diff
@@ -0,0 +1,81 @@
+module SyncChecker
+  module Checks
+    class TopicsCheck
+      attr_reader :publishing_api_client, :edition, :content_item
+
+      def initialize(edition, publishing_api_client = Whitehall.publishing_api_v2_client)
+        @edition = edition
+        @publishing_api_client = publishing_api_client
+      end
+
+      def call(response)
+        failures = []
+        if response.response_code == 200
+          @content_item = JSON.parse(response.body)
+          failures << check_parent
+          failures << check_topics
+        end
+        failures.flatten.compact
+      end
+
+    private
+
+      def check_parent
+        return if links_parent.nil? && parent_base_path.nil?
+        return "expected links#parent but it isn't present" if links_parent.nil? && parent_base_path.present?
+        return "links#parent is present but shouldn't be" if links_parent.present? && parent_base_path.blank?
+
+        parent_content_id = links_parent["content_id"]
+        expected_parent_content_id = content_ids["/topic/#{edition.primary_specialist_sector_tag}"]
+        if parent_content_id != expected_parent_content_id
+          "expected parent#content_id to be '#{expected_parent_content_id}' but got '#{parent_content_id}'"
+        end
+      end
+
+      def check_topics
+        return if topic_base_paths.empty? && links_topics.nil?
+        return "expected links#topics but it isn't present" if links_topics.nil? && topic_base_paths.any?
+        return "links#topics are present but shouldn't be" if links_topics.present? && topic_base_paths.empty?
+        check_for_missing_topics + check_for_unexpected_topics
+      end
+
+      def check_for_missing_topics
+        expected_ids = content_ids.values
+        expected_ids
+          .reject { |content_id| links_topics_content_ids.include?(content_id)}
+          .map { |missing_content_id| "links#topics should contain '#{missing_content_id}' but doesn't" }
+      end
+
+      def check_for_unexpected_topics
+        links_topics_content_ids
+          .reject { |content_id| content_ids.values.include?(content_id)}
+          .map { |unexpected_content_id| "links#topics contains '#{unexpected_content_id}' but shouldn't" }
+      end
+
+      def content_ids
+        @content_ids ||= publishing_api_client.lookup_content_ids(base_paths: topic_base_paths)
```

Comment:
As this is going to all be used within a single run of a single Ruby process, we could memoize the fetch of these. There's only 376 distinct tags, so storing a Hash of base path to content id won't take much memory, and will massively reduce round trips to Publishing API. We could even pre-load them all if we wanted.